mk/config.mk

```diff
@@ -350,13 +350,13 @@ CFG_CORE_NEX_HEAP_SIZE ?= 16384
 # tee-supplicant)
 CFG_TA_GPROF_SUPPORT ?= n
 
-# Enable to compile user TA libraries with profiling (-pg).
+# Enable to compile user TA libraries with cflag (-pg).
 # Depends on CFG_TA_GPROF_SUPPORT.
-CFG_ULIBS_GPROF ?= n
+CFG_ULIBS_INSTRUMENT ?= n
 
-ifeq ($(CFG_ULIBS_GPROF),y)
-ifneq ($(CFG_TA_GPROF_SUPPORT),y)
-$(error Cannot instrument user libraries if user mode profiling is disabled)
+ifeq ($(CFG_ULIBS_INSTRUMENT),y)
+ifeq ($(CFG_TA_GPROF_SUPPORT),n)
+$(error Cannot instrument user libs if user mode profiling is disabled)
```

Comment:
It seems that one of the reason being unnecessarily filling the TA `.bss` section with gprof and ftrace buffers which remain unused in case only `CFG_ULIBS_INSTRUMENT=y` is enabled. See `ta/arch/arm/ta.ld.S`.